nucypher/blockchain/eth/sol/source/contracts/staking_contracts/WorkLockPoolingContract.sol

```diff
@@ -0,0 +1,439 @@
+        delegator.depositedTokens = delegator.depositedTokens.add(_value);
+        token.safeTransferFrom(msg.sender, address(this), _value);
+        emit TokensDeposited(msg.sender, _value, delegator.depositedTokens);
+    }
+
+    /**
+     * @notice Delegator can transfer ETH directly to workLock
+     */
+    function escrowETH() external payable {
+        Delegator storage delegator = delegators[msg.sender];
+        delegator.depositedETHWorkLock = delegator.depositedETHWorkLock.add(msg.value);
+        totalWorkLockETHReceived = totalWorkLockETHReceived.add(msg.value);
+        workLock.bid{value: msg.value}();
+        emit Bid(msg.sender, msg.value);
+    }
+
+    /**
+     * @dev Hide method from StakingInterface
+     */
+    function bid(uint256) public payable {
+        revert();
+    }
+
+    /**
+     * @dev Hide method from StakingInterface
+     */
+    function withdrawCompensation() public pure {
+        revert();
+    }
+
+    /**
+     * @dev Hide method from StakingInterface
+     */
+    function cancelBid() public pure {
+        revert();
+    }
+
+    /**
+     * @dev Hide method from StakingInterface
+     */
+    function claim() public pure {
+        revert();
+    }
+
+    /**
+     * @notice Claim tokens in WorkLock and save number of claimed tokens
+     */
+    function claimTokensFromWorkLock() public {
+        workLockClaimedTokens = workLock.claim();
+        totalDepositedTokens = totalDepositedTokens.add(workLockClaimedTokens);
+        emit Claimed(address(this), workLockClaimedTokens);
+    }
+
+    /**
+     * @notice Calculate and save number of claimed tokens for specified delegator
+     */
+    function calculateAndSaveTokensAmount() external {
+        Delegator storage delegator = delegators[msg.sender];
+        calculateAndSaveTokensAmount(delegator);
+    }
+
+    /**
+     * @notice Calculate and save number of claimed tokens for specified delegator
+     */
+    function calculateAndSaveTokensAmount(Delegator storage _delegator) internal {
+        if (workLockClaimedTokens == 0 ||
+            _delegator.depositedETHWorkLock == 0 ||
+            _delegator.claimedWorkLockTokens)
+        {
+            return;
+        }
+
+        uint256 delegatorTokensShare = _delegator.depositedETHWorkLock.mul(workLockClaimedTokens)
+            .div(totalWorkLockETHReceived);
+
+        _delegator.depositedTokens = _delegator.depositedTokens.add(delegatorTokensShare);
+        _delegator.claimedWorkLockTokens = true;
+        emit Claimed(msg.sender, delegatorTokensShare);
+    }
+
+    /**
+     * @notice Get available reward for all delegators and owner
+     */
+    function getAvailableReward() public view returns (uint256) {
+        uint256 stakedTokens = escrow.getAllTokens(address(this));
+        uint256 freeTokens = token.balanceOf(address(this));
+        uint256 reward = stakedTokens.add(freeTokens).sub(totalDepositedTokens);
+        if (reward > freeTokens) {
```

Comment:
```suggestion
        if (stakedTokens > totalDepositedTokens) {
```
This should be equivalent, but I find it more clear, since it's very easy to understand that this covers the case when you have more tokens in StakingEscrow that you deposited in the first place.

Reply:
Yeah, that's equivalent, for me current option is more clear because I see that any reward can't be more than contract has. What others will say?